internal/controller/rbac/provider/roles/reconciler.go

```diff
@@ -198,6 +250,28 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 		}
 	}
 
+	rejected, err := r.rbac.ValidatePermissionRequests(ctx, pr.Spec.PermissionRequests...)
+	if err != nil {
+		log.Debug(errValidatePermissions, "error", err)
+		r.record.Event(pr, event.Warning(reasonApplyRoles, errors.Wrap(err, errValidatePermissions)))
+		return reconcile.Result{RequeueAfter: shortWait}, nil
+	}
+
+	for _, rule := range rejected {
+		log.Debug(errRejectedPermission, "rule", rule)
+		r.record.Event(pr, event.Warning(reasonApplyRoles, errors.Errorf("%s %s", errRejectedPermission, rule)))
+	}
+
+	// We return early and don't grant _any_ RBAC permissions if we would reject
+	// any requested permission. It's better for the provider to be completely
+	// and obviously broken than for it to be subtly broken in a way that may
+	// not surface immediately, i.e. due to missing an RBAC permission it only
+	// occasionally needs. There's no need to requeue - the revisions requests
+	// won't change, and we're watching the ClusterRole of allowed requests.
```

Comment:
Somewhat related: how would you feel about us updating providers to panic if types are missing or we don't have RBAC? It can be confusing when the pods are running normally but in reality they can't perform any operations.

Reply:
Yes, we should refuse to run if we know we can't run properly. I thought this was the case already when CRDs were missing, but I know we don't crash for missing RBAC permissions.